src/Routing/Middleware/AssetMiddleware.php

```diff
@@ -54,30 +57,29 @@ public function __construct(array $options = [])
      * Serve assets if the path matches one.
      *
      * @param \Psr\Http\Message\ServerRequestInterface $request The request.
-     * @param \Cake\Http\Response $response The response.
-     * @param callable $next Callback to invoke the next middleware.
-     * @return \Cake\Http\Response A response
+     * @param \Psr\Http\Server\RequestHandlerInterface $handler The request handler.
+     * @return \Psr\Http\Message\ResponseInterface A response.
      */
-    public function __invoke(ServerRequestInterface $request, Response $response, callable $next): Response
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
         $url = $request->getUri()->getPath();
         if (strpos($url, '..') !== false || strpos($url, '.') === false) {
-            return $next($request, $response);
+            return $handler->handle($request);
         }
 
         if (strpos($url, '/.') !== false) {
-            return $next($request, $response);
+            return $handler->handle($request);
         }
 
         $assetFile = $this->_getAssetFile($url);
         if ($assetFile === null || !file_exists($assetFile)) {
-            return $next($request, $response);
+            return $handler->handle($request);
         }
 
         $file = new SplFileInfo($assetFile);
         $modifiedTime = $file->getMTime();
         if ($this->isNotModified($request, $file)) {
-            return $response
+            return (new Response())
                 ->withStringBody('')
```

Comment:
With fresh `Response`, you are now sure body is empty, so resetting body unnecessary.